File: DataTypes.py
import numpy as np
from pyspark.mllib.linalg import Vectors
from pyspark.mllib.regression import LabeledPoint
from pyspark.sql import SparkSession

if __name__ == "__main__":
    sparkSession = SparkSession\
        .builder\
        .getOrCreate()

    # Python list
    dense_vector1 = [1.0, 0.0, 3.5, 0.0, 5.1]

    # NumPy array
    dense_vector2 = np.array([1.0, 0.0, 3.5, 0.0, 5.1])

    # Vector
    dense_vector3 = Vectors.dense([1.0, 0.0, 3.5, 0.0, 5.1])

    sparse_vector = Vectors.sparse(5, [0, 2, 4], [1.0, 3.5, 5.1])

    print("Vector 1 (Python list) : " + str(dense_vector1))
    print("Vector 2 (NumPy Array) : " + str(dense_vector2))
    print("Vector 3 (Vectors) : " + str(dense_vector3))
    print("Vector 1 (Vectors): " + str(sparse_vector))

    labeled_point = LabeledPoint(1.0, dense_vector1)
    labeled_point2 = LabeledPoint(0.0, Vectors.sparse(5, [2, 4], [5.2, 6.2]))

    print("Labeled point (Python list): " + str(labeled_point))
    print("Labeled point (Sparse vector): " + str(labeled_point2))

    sparkSession.stop()

File: Java/Java/DataTypes.java
import org.apache.log4j.Level;
import org.apache.log4j.Logger;
import org.apache.spark.mllib.linalg.DenseVector;
import org.apache.spark.mllib.linalg.SparseVector;
import org.apache.spark.mllib.linalg.Vector;
import org.apache.spark.mllib.linalg.Vectors;
import org.apache.spark.mllib.regression.LabeledPoint;
import org.apache.spark.sql.SparkSession;

import java.io.IOException;

public class DataTypes {
  public static void main(String[] args) {
    Logger.getLogger("org").setLevel(Level.OFF) ;

    // Create a SparkSession.
    SparkSession sparkSession = SparkSession
            .builder()
            .appName("SparkML datatypes")
            .master("local[2]")
            .getOrCreate();

    Vector denseVector = new DenseVector(new double[]{1.0, 0.0, 3.2}) ;
    Vector denseVector2 = Vectors.dense(1.0, 2.0, 3.0, 4.0, 5.0);

    Vector sparseVector = new SparseVector(3, new int[]{0, 2}, new double[]{1.0, 3.5}) ;
    Vector sparseVector2 = Vectors.sparse(3, new int[]{0, 2}, new double[]{1.0, 3.5}) ;

    System.out.println("Vector 1 (dense): " + denseVector) ;
    System.out.println("Vector 2 (dense): " + denseVector2) ;
    System.out.println("Vector 1 (sparse): " + sparseVector) ;
    System.out.println("Vector 2 (sparse): " + sparseVector2) ;

    LabeledPoint labeledPoint = new LabeledPoint(1.0, denseVector) ;
    LabeledPoint labeledPoint2 = new LabeledPoint(0.0, Vectors.sparse(5, new int[]{2, 4}, new double[]{5.2, 6.2})) ;

    System.out.println("Labeled point 1: " + labeledPoint) ;
    System.out.println("Labeled point 2: " + labeledPoint2) ;

    sparkSession.stop();
  }
}
